--- a/core/src/avm1/globals/color_matrix_filter.rs
+++ b/core/src/avm1/globals/color_matrix_filter.rs
@@ -61,9 +61,8 @@ impl<'gc> ColorMatrixFilter<'gc> {
     }
 
     fn matrix(self, activation: &mut Activation<'_, 'gc>) -> Value<'gc> {
-        // Use `.as_array_of_cells()` to avoid a copy out of the `Cell`.
-        let matrix = self.0.matrix.as_array_of_cells();
+        let matrix = self.0.matrix.get();
 
         ArrayBuilder::new(activation)
-            .with(matrix.iter().map(|v| v.get().into()))
+            .with(matrix.iter().copied().map(|v| v.into()))
             .into()
     }
