diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -156,7 +156,22 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-rgb30.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-rgb30-v2.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-rgb20pro.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-rk2023.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-powkiddy-x55.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-powkiddy-x35h.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-powkiddy-x35s.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-powkiddy-rk2023.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-powkiddy-rgb20-pro.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-powkiddy-rgb20sx.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-powkiddy-rgb30.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-powkiddy-rgb10max3.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-gamemt-e5p.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-gamemt-e6p.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-diium-d50plus.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-coolboy-h9.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-miniloong.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-miyoo-flip.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-radxa-zero-3w.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3567-evb2-lp4x-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3567-evb2-lp4x-v10-dual-channel-lvds.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3567-evb2-lp4x-v10-one-vp-two-single-channel-lvds.dtb
