diff -Naur a/product/kernel/drivers/gpu/arm/midgard/mali_kbase_native_mgm.c b/product/kernel/drivers/gpu/arm/midgard/mali_kbase_native_mgm.c
--- a/product/kernel/drivers/gpu/arm/midgard/mali_kbase_native_mgm.c	2026-03-16 02:44:15.000000000 +0800
+++ b/product/kernel/drivers/gpu/arm/midgard/mali_kbase_native_mgm.c	2026-07-03 17:30:33.327056646 +0800
@@ -122,12 +122,16 @@
 }
 
 static u64 kbase_native_mgm_update_gpu_pte(struct memory_group_manager_device *mgm_dev,
-					   unsigned int group_id, int mmu_level, u64 pte)
+					   unsigned int group_id, unsigned int pbha_id,
+					   unsigned int pte_flags, int mmu_level, u64 pte)
 {
 	if (WARN_ON(group_id >= MEMORY_GROUP_MANAGER_NR_GROUPS))
 		return pte;
 
-	pte |= ((u64)group_id << PTE_PBHA_SHIFT) & PTE_PBHA_MASK;
+	if ((pte_flags & BIT(MMA_VIOLATION)) && pbha_id) {
+		pr_warn_once("MMA violation! Applying PBHA override workaround to PTE\n");
+		pte |= ((u64)pbha_id << PTE_PBHA_SHIFT) & PTE_PBHA_MASK;
+	}
 
 	/* Address could be translated into a different bus address here */
 	pte |= ((u64)1 << PTE_RES_BIT_MULTI_AS_SHIFT);
@@ -135,6 +139,16 @@
 	return pte;
 }
 
+static bool kbase_native_mgm_get_import_memory_cached_access_permitted(
+	struct memory_group_manager_device *mgm_dev,
+	struct memory_group_manager_import_data *import_data)
+{
+	CSTD_UNUSED(mgm_dev);
+	CSTD_UNUSED(import_data);
+
+	return true;
+}
+
 static u64 kbase_native_mgm_pte_to_original_pte(struct memory_group_manager_device *mgm_dev,
 						unsigned int group_id, int mmu_level, u64 pte)
 {
@@ -158,6 +172,8 @@
 		.mgm_vmf_insert_pfn_prot = kbase_native_mgm_vmf_insert_pfn_prot,
 		.mgm_update_gpu_pte = kbase_native_mgm_update_gpu_pte,
 		.mgm_pte_to_original_pte = kbase_native_mgm_pte_to_original_pte,
+		.mgm_get_import_memory_cached_access_permitted =
+			kbase_native_mgm_get_import_memory_cached_access_permitted,
 	},
 	.data = NULL
 };
diff -Naur a/product/kernel/drivers/gpu/arm/midgard/mmu/mali_kbase_mmu.c b/product/kernel/drivers/gpu/arm/midgard/mmu/mali_kbase_mmu.c
--- a/product/kernel/drivers/gpu/arm/midgard/mmu/mali_kbase_mmu.c	2026-03-16 02:44:15.000000000 +0800
+++ b/product/kernel/drivers/gpu/arm/midgard/mmu/mali_kbase_mmu.c	2026-07-03 17:31:53.487724995 +0800
@@ -38,6 +38,7 @@
 #include <mmu/mali_kbase_mmu_hw.h>
 #include <mali_kbase_mem.h>
 #include <mali_kbase_reset_gpu.h>
+#include <linux/memory_group_manager.h>
 #include <mmu/mali_kbase_mmu.h>
 #include <mmu/mali_kbase_mmu_internal.h>
 #include <device/mali_kbase_device.h>
@@ -2217,7 +2218,8 @@
 
 		kbdev->mmu_mode->entry_set_pte(&pte, target_pgd);
 		parent_page_va[parent_vpfn] = kbdev->mgm_dev->ops.mgm_update_gpu_pte(
-			kbdev->mgm_dev, MGM_DEFAULT_PTE_GROUP, parent_index, pte);
+			kbdev->mgm_dev, MGM_DEFAULT_PTE_GROUP, PBHA_ID_DEFAULT, PTE_FLAGS_NONE,
+			parent_index, pte);
 		kbdev->mmu_mode->set_num_valid_entries(parent_page_va, current_valid_entries + 1);
 		kunmap_pgd(parent_page, parent_page_va);
 
@@ -2677,10 +2679,15 @@
 			 unsigned long const flags, int const level, int const group_id)
 {
 	u64 entry;
+	unsigned int pte_flags = 0;
 
 	kbdev->mmu_mode->entry_set_ate(&entry, phy, flags, level);
-	return kbdev->mgm_dev->ops.mgm_update_gpu_pte(kbdev->mgm_dev, (unsigned int)group_id, level,
-						      entry);
+
+	if ((flags & KBASE_REG_GPU_CACHED) && !(flags & KBASE_REG_CPU_CACHED))
+		pte_flags |= BIT(MMA_VIOLATION);
+
+	return kbdev->mgm_dev->ops.mgm_update_gpu_pte(kbdev->mgm_dev, (unsigned int)group_id, 0,
+						      pte_flags, level, entry);
 }
 
 static int mmu_insert_pages_no_flush(struct kbase_device *kbdev, struct kbase_mmu_table *mmut,
@@ -3800,7 +3807,8 @@
 	kbdev->mmu_mode->entry_set_pte(&managed_pte, old_pgd_phys);
 	target = &parent_pgd_page[index];
 	*target = kbdev->mgm_dev->ops.mgm_update_gpu_pte(kbdev->mgm_dev, MGM_DEFAULT_PTE_GROUP,
-							 level, managed_pte);
+							 PBHA_ID_DEFAULT, PTE_FLAGS_NONE, level,
+							 managed_pte);
 
 	kbdev->mmu_mode->set_num_valid_entries(parent_pgd_page, num_of_valid_entries);
 	kunmap_atomic_pgd(parent_pgd_page);
@@ -3981,7 +3989,8 @@
 #endif
 	kbdev->mmu_mode->entry_set_pte(&managed_pte, new_pgd_phys);
 	*target = kbdev->mgm_dev->ops.mgm_update_gpu_pte(kbdev->mgm_dev, MGM_DEFAULT_PTE_GROUP,
-							 level, managed_pte);
+							 PBHA_ID_DEFAULT, PTE_FLAGS_NONE, level,
+							 managed_pte);
 
 	kbdev->mmu_mode->set_num_valid_entries(parent_pgd_page, num_of_valid_entries);
 
