diff -Naur a/product/kernel/drivers/gpu/arm/midgard/backend/gpu/mali_kbase_devfreq.c b/product/kernel/drivers/gpu/arm/midgard/backend/gpu/mali_kbase_devfreq.c
--- a/product/kernel/drivers/gpu/arm/midgard/backend/gpu/mali_kbase_devfreq.c	2024-10-02 19:09:46.000000000 +0800
+++ b/product/kernel/drivers/gpu/arm/midgard/backend/gpu/mali_kbase_devfreq.c	2026-07-03 10:03:31.156843456 +0800
@@ -140,8 +140,8 @@
 #endif
 #if KERNEL_VERSION(4, 15, 0) <= LINUX_VERSION_CODE
 	err = dev_pm_genpd_set_performance_state(kbdev->dev, nominal_freq);
-	/* For ENODEV or EOPNOTSUPP do not return error code */
-	if (err && !((err == -ENODEV) || (err == -EOPNOTSUPP))) {
+	/* Rockchip genpd does not implement performance states; ignore EINVAL too. */
+	if (err && !((err == -ENODEV) || (err == -EOPNOTSUPP) || (err == -EINVAL))) {
 		dev_err(dev, "Failed to set opp (%d) (target %lu)\n", err, *target_freq);
 		return err;
 	}
