diff --git a/drivers/mmc/core/host.c b/drivers/mmc/core/host.c
index 190f308c0e0f..f3fc8b01dab3 100644
--- a/drivers/mmc/core/host.c
+++ b/drivers/mmc/core/host.c
@@ -406,6 +406,8 @@ int mmc_of_parse(struct mmc_host *host)
 		host->caps2 |= MMC_CAP2_NO_SD;
 	if (device_property_read_bool(dev, "no-mmc"))
 		host->caps2 |= MMC_CAP2_NO_MMC;
+	if (device_property_read_bool(dev, "supports-rk912"))
+		host->caps2 |= MMC_CAP2_WIFI_RK912;
 	if (device_property_read_bool(dev, "no-mmc-hs400"))
 		host->caps2 &= ~(MMC_CAP2_HS400_1_8V | MMC_CAP2_HS400_1_2V |
 				 MMC_CAP2_HS400_ES);
diff --git a/drivers/mmc/core/sdio.c b/drivers/mmc/core/sdio.c
index cb87e8273779..5e9d51c9d3c0 100644
--- a/drivers/mmc/core/sdio.c
+++ b/drivers/mmc/core/sdio.c
@@ -1076,7 +1076,8 @@ static int mmc_sdio_resume(struct mmc_host *host)
 	 * removable card is checked from a detect work later on in the resume
 	 * process.
 	 */
-	if (!mmc_card_keep_power(host)) {
+	if (!mmc_card_keep_power(host) &&
+	    !(host->caps2 & MMC_CAP2_WIFI_RK912)) {
 		mmc_power_up(host, host->card->ocr);
 		/*
 		 * Tell runtime PM core we just powered up the card,
diff --git a/drivers/mmc/core/sdio_cis.c b/drivers/mmc/core/sdio_cis.c
index afaa6cab1adc..e614c5c0ddd7 100644
--- a/drivers/mmc/core/sdio_cis.c
+++ b/drivers/mmc/core/sdio_cis.c
@@ -190,7 +190,7 @@ static int cistpl_funce_func(struct mmc_card *card, struct sdio_func *func,
 		pr_warn("%s: card has broken SDIO 1.1 CIS, forcing SDIO 1.0\n",
 			mmc_hostname(card->host));
 		vsn = SDIO_SDIO_REV_1_00;
-	} else if (size < min_size) {
+	} else if (size < min_size && !(card->host->caps2 & MMC_CAP2_WIFI_RK912)) {
 		return -EINVAL;
 	}
 
@@ -198,7 +198,7 @@ static int cistpl_funce_func(struct mmc_card *card, struct sdio_func *func,
 	func->max_blksize = buf[12] | (buf[13] << 8);
 
 	/* TPLFE_ENABLE_TIMEOUT_VAL, present in ver 1.1 and above */
-	if (vsn > SDIO_SDIO_REV_1_00)
+	if (vsn > SDIO_SDIO_REV_1_00 && !(card->host->caps2 & MMC_CAP2_WIFI_RK912))
 		func->enable_timeout = (buf[28] | (buf[29] << 8)) * 10;
 	else
 		func->enable_timeout = jiffies_to_msecs(HZ);
diff --git a/drivers/mmc/host/dw_mmc.c b/drivers/mmc/host/dw_mmc.c
index a9aaafefbd93..a79f88de8f3d 100644
--- a/drivers/mmc/host/dw_mmc.c
+++ b/drivers/mmc/host/dw_mmc.c
@@ -326,6 +326,10 @@ static u32 dw_mci_prepare_command(struct mmc_host *mmc, struct mmc_command *cmd)
 			     SDMMC_CMD_PRV_DAT_WAIT, 0);
 	}
 
+	if ((mmc->caps2 & MMC_CAP2_WIFI_RK912) &&
+	    (cmd->opcode == SD_IO_RW_DIRECT))
+		cmdr |= SDMMC_CMD_PRV_DAT_WAIT;
+
 	if (cmd->flags & MMC_RSP_PRESENT) {
 		/* We expect a response, so set this bit */
 		cmdr |= SDMMC_CMD_RESP_EXP;
@@ -1310,7 +1314,8 @@ static void dw_mci_setup_bus(struct dw_mci_slot *slot, bool force_clkinit)
 
 		/* enable clock; only low power if no SDIO */
 		clk_en_a = SDMMC_CLKEN_ENABLE << slot->id;
-		if (!test_bit(DW_MMC_CARD_NO_LOW_PWR, &slot->flags))
+		if (!test_bit(DW_MMC_CARD_NO_LOW_PWR, &slot->flags) &&
+		    !(slot->mmc->caps2 & MMC_CAP2_WIFI_RK912))
 			clk_en_a |= SDMMC_CLKEN_LOW_PWR << slot->id;
 		mci_writel(host, CLKENA, clk_en_a);
 
diff --git a/include/linux/mmc/host.h b/include/linux/mmc/host.h
index 62a6847a3b6f..a596aa4e276e 100644
--- a/include/linux/mmc/host.h
+++ b/include/linux/mmc/host.h
@@ -427,6 +427,7 @@ struct mmc_host {
 #define MMC_CAP2_CRYPTO		0
 #endif
 #define MMC_CAP2_ALT_GPT_TEGRA	(1 << 28)	/* Host with eMMC that has GPT entry at a non-standard location */
+#define MMC_CAP2_WIFI_RK912     (1 << 29)
 
 	int			fixed_drv_type;	/* fixed driver type for non-removable media */
 
